src/github/GithubClient.ts

```diff
@@ -0,0 +1,368 @@
+import { create as createArtifactClient } from "@actions/artifact";
+import { DownloadHttpClient } from "@actions/artifact/lib/internal/download-http-client";
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { GitHub } from "@actions/github/lib/utils";
+import * as cache from "@actions/tool-cache";
+import { Release } from "@octokit/webhooks-types";
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+export type GithubRepo = { owner: string; repo: string };
+
+/**
+ * Basic release asset information
+ */
+export interface ReleaseAsset {
+  id: number;
+  name: string;
+}
+
+/**
+ * Common interface for methods requiring a release
+ */
+interface ReleaseProps {
+  release: Release;
+}
+
+/**
+ * Basic artifact interface returned via listWorkflowArtifacts
+ */
+export interface Artifact {
+  name: string;
+}
+
+/**
+ * Basic workflow run information
+ */
+export interface WorkflowRun {
+  id: number;
+}
+
+/**
+ * Provides a basic shim to interact with the necessary Github APIs
+ */
+export class GithubClient {
+  client: InstanceType<typeof GitHub>;
+  repo: GithubRepo;
+
+  constructor(client: InstanceType<typeof GitHub>, repo: GithubRepo) {
+    this.client = client;
+    this.repo = repo;
+  }
+
+  // --------------- WORKFLOW ARTIFACT METHODS ------------------
+
+  /**
+   * Lists the workflow artifacts for the current workflow
+   */
+  async listWorkflowArtifacts(): Promise<Artifact[]> {
+    // The REST listWorkflowRunArtifacts endpoint does not seem to work during
+    // the workflow run, presumably the are available afterwards much like the
+    // Github UI only shows artifacts after completion of a run, so we have
+    // to do a little bit of hackery here. We _could_ download all artifacts
+    // using a supported API, but internally it's using this anyway
+    const downloadClient = new DownloadHttpClient();
+    const response = await downloadClient.listArtifacts();
+
+    core.debug(dashWrap("listWorkflowArtifacts"));
+    core.debug(JSON.stringify(response));
+
+    return response.value;
+  }
+
+  /**
+   * Downloads a workflow artifact for the current workflow run
+   * @param name artifact name
+   */
+  async downloadWorkflowArtifact({ name }: { name: string }): Promise<string> {
+    const client = createArtifactClient();
+    const tempPath = fs.mkdtempSync(path.join(os.tmpdir(), "sbom-action-"));
```

Comment:
Do we want the trailing - on `sbom-action-` here? I don't think we ever reference it again by it's value since it becomes `response.downloadPath`, I was just curious if github appends anything which is why we would put `-` at the end.

Reply:
this has to do with directory naming. e.g. mktmpdir creates a directory `<prefix><randomHash>` so this makes things that look a little better like `sbom-action-a45fd923`

Reply:
Nice - thanks for the clarification! I didn't even notice mkdtemp would add the hash at the end.